formulaic/transforms/basis_spline.py

```diff
@@ -141,7 +148,7 @@ def basis_spline(  # pylint: disable=dangerous-default-value  # always replaced
             )
         knots.insert(0, cast(float, lower_bound))
         knots.append(cast(float, upper_bound))
-        knots = list(numpy.pad(knots, degree, mode="edge"))
+        knots = numpy.pad(knots, degree, mode="edge").tolist()
```

Comment:
The issue here is that `list(array([1.0, 2.0]))` is `[np.float64(1.0), np.float64(2.0)]`, which isn't wrong per-se, but `tolist()` return native python types.

Reply:
Using this in 2 places in #221 now. `tolist` is more performant than `list(arr)` since NumPy knows about the layout of the data, and so can iterate in C.